datafusion/core/src/optimizer/simplify_expressions.rs

```diff
@@ -375,7 +375,9 @@ impl<'a> ConstEvaluator<'a> {
             | Expr::AggregateUDF { .. }
             | Expr::ScalarVariable(_, _)
             | Expr::Column(_)
-            | Expr::Exists(_)
+            | Expr::Exists { .. }
+            | Expr::InSubquery { .. }
+            | Expr::ScalarSubquery(_)
```

Comment:
at some point it would be cool to be able to evaluate scalar subqueries at plan time, "but not now" 😆